oas3.yaml

```diff
@@ -0,0 +1,328 @@
+openapi: 3.0.1
+info:
+  title: Shelter API
+  description: |
+    A website which provides information about pets which are being sheltered and are available
+    for adoption.
+
+  license:
+    name: MIT
+  version: 1.0.0
+
+servers:
+  - url: http://localhost:5000
+  - url: https://localhost:5001
+
+tags:
+  - name: health
+    description: APIs used to determine whether your service is healthy and running as expected.
+
+  - name: pets
+    description: APIs used to enumerate and manage the collection of pets which are available for adoption.
+
+paths:
+  /api/v1/health:
+    get:
+      tags:
+        - health
+
+      summary: Get Health (v1)
+      description: Gets information about the current health state of your service.
+      operationId: health_v1
+      responses:
+        200:
+          description: The service is healthy.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/HealthV1"
+              example:
+                status: Healthy
+                results:
+                  database:
+                    status: Healthy
+                    description: Connected to upstream database server.
+                    data: {}
+        500:
+          description: The service is unhealthy.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/HealthV1"
+              example:
+                status: Unhealthy
+                results:
+                  database:
+                    status: Unhealthy
+                    description: Unable to connect to the upstream database server.
+                    data: {}
+
+  /api/v1/pets:
+    get:
```

Comment:
this one is not much scalable - but ours is a small pet shelter after all :)
jokes aside, it can be left like this, as Nick was suggesting, to spark some discussion on scalability, pagination

Reply:
Very good points, I had opted to leave pagination out for now just because it introduces quite a bit of added complexity in the backend and UI to support it - however if we were to make this very scaleable then that's absolutely something we'd want to add.

Reply:
I doubt if we'll have a huge amount of time to do anything very complex in the UI to take advantage of this, however from an api design point of view, it might be useful to have filtering and paging in here in order to show what a good design may include.  Since this is essentially a specification, we could use these as talking points without expecting that the candidates have to necessarily implement all of it.

Reply:
Yeah, that's a good idea, how does it sound for us to maybe add that as a `v2` endpoint and tell candidates that they can implement it later as a stretch goal (if they want to)? Could illustrate how an API might evolve over the lifespan of the service and let us talk about API versioning and its role in long term maintainability.

Reply:
This won't fit into the classroom but filtering and paging is super easy with odata. Maybe we can mention it as "extended reading"?

Reply:
Sounds good to me.

Reply:
I think that's a great idea, OData is definitely an interesting option for them to look into